docs/md-loader.js

```diff
@@ -1,23 +1,47 @@
+// @ts-check
 const GrayMatter = require('gray-matter');
+const Unified = require('unified');
+const RemarkParse = require('remark-parse');
+const Gfm = require('remark-gfm');
+const RemarkRehype = require('remark-rehype');
+const RehypeStringify = require('rehype-stringify');
+const RehypeRaw = require('rehype-raw');
 
-// makes mdx in next.js suck less by injecting necessary exports so that
-// the docs are still readable on github
-// (Shamelessly stolen from React Query docs)
-// (Which was shamelessly stolen from Expo.io docs)
-// @see https://github.com/tannerlinsley/react-query/blob/16b7d290c70639b627d9ada32951d211eac3adc3/docs/src/lib/docs/md-loader.js
-// @see https://github.com/expo/expo/blob/303cb7b689603223401c091c6a2e1e01f182d355/docs/common/md-loader.js
+/**
+ * @param {string} source
+ * @param {import('unified').PluggableList} plugins
+ * @returns {import('vfile').VFile}
+ */
+function toHtmlString(source, plugins) {
+  const processor = Unified()
+    .use(RemarkParse)
+    .use(plugins)
+    .use(RemarkRehype, { allowDangerousHtml: true })
+    .use(RehypeRaw)
+    .use(RehypeStringify);
 
-module.exports = function addLayoutToMdx(source) {
+  return processor.processSync(source);
+}
+
+/**
+ * @type {import('webpack').LoaderDefinitionFunction<{remarkPlugins: import('unified').PluggableList}>}
+ * @this {import('webpack').LoaderContext<{remarkPlugins: import('unified').PluggableList}>}
+ */
+module.exports = function loadMarkdownWithHtml(source, sourceMap, additionalData) {
   const callback = this.async();
 
   const { content, data } = GrayMatter(source);
-  const code =
-    `import { Layout } from '/src/components/Layout';
+
+  const html = toHtmlString(content, this.getOptions().remarkPlugins).contents;
+
+  const code = `
+import { Layout } from '/src/components/Layout';
 export const meta = ${JSON.stringify(data)};
-export default ({ children, ...props }) => (
-  <Layout meta={meta} {...props}>{children}</Layout>
+const Page = ({ children, ...props }) => (
+  <Layout meta={meta} {...props}><div dangerouslySetInnerHTML={{__html: ${JSON.stringify(html)} }} /></Layout>
 );
-` + content.replace(/<!-- (.*?) -->/g, '');
+export default Page;
+`.trim();
```

Comment:
Interesting, I have two questions,
1) We have two files with similar content `md-loader.js` & `md-layout.js`, what's the difference between them? I see that one uses children property and the other sets inner html.
2) I suppose that the `code` variable content is somehow injected into the `/pages` directory during build for every `article`?

Reply:
`md-loader.js` is used for injecting HTML generated from markdown files generated from api-extractor.

`md-layout.js` will be used for .mdx files we write by hand